File: CargoFlash.ROMS.Model/Roster/DutyArea.cs
using System.Runtime.Serialization;
using System;

namespace CargoFlash.Cargo.Model.Roster
{
    [KnownType(typeof(DutyArea))]
    public class DutyArea
    {
        public int SNo { get; set; }
        public string AreaName { get; set; }
        public bool IsActive { get; set; }
        public string CreatedOn { get; set; }
        public string CreatedBy { get; set; }
        public string UpdatedBy { get; set; }
        public string UpdatedOn { get; set; }
       
        public string Active { get; set; }
        public string ColorName { get; set; }
        public int TerminalSno { get; set; }
        public string Text_TerminalSno { get; set; }
        public bool IsExport { get; set; }
        public string Export { get; set; }
        public string DutyAreaName { get; set; }
        public int HashColorCodeSno { get; set; }
        public string Text_HashColorCodeSno { get; set; }
    }

    [KnownType(typeof(HashColorName))]
    public class HashColorName
    {
        public string ColorName { get; set; }
    }
}

File: CargoFlashCargoWebApps/JScript/Roster/DutyArea.js
$(document).ready(function () {
    var hashColorCodeSno = ''; var hasTerminalSNo = '';
    cfi.ValidateForm();
    cfi.AutoCompleteV2("TerminalSno", "SNo,TerminalName", "Roster_TerminalName",  OnSelectTerminal, "contains");
    cfi.AutoCompleteV2("HashColorCodeSno", "HashColorName", "Roster_HashColorName",OnSelectHashColorCode, "contains");
});

hashColorCodeSno = $("#HashColorCodeSno").val();

function OnSelectHashColorCode() {
    hashColorCodeSno = ($("#HashColorCodeSno").val() == "") ? "0" : $("#HashColorCodeSno").val();
    try {
        $.ajax({
            type: "GET",
            url: "./Services/Roster/DutyAreaService.svc/GetColorName/" + hashColorCodeSno,
            dataType: "json",
            success: function (response) {
                $("#ColorName").val(response.ColorName);
                $("span#ColorName").html(response.ColorName);
            }
        });
    }
    catch (exp) { }
}

function BindDutyAreaName() {

    var TypeCheck =$('#IsExport:checked').val();
    var Type = (TypeCheck == "0") ? "IMPORT" : "EXPORT";
    var TerminalName = $("#Text_TerminalSno").val();
    var areaName = $('#AreaName').val().toUpperCase();
    if (areaName == "" || TerminalName == "") {
        $("#DutyAreaName").val('');
        $("span#DutyAreaName").html('');
    }
    else {
        $("#DutyAreaName").val(TerminalName + ' - ' + Type + ' - ' + areaName);
        $("span#DutyAreaName").html(TerminalName + ' - ' + Type + ' - ' + areaName);
    }
}

function OnSelectTerminal() {
    BindDutyAreaName();
}

$("#AreaName").bind("change", OnSelectTerminal);

$('input[name="IsExport"]:radio').change(function () {
    BindDutyAreaName();
});
